boards/riscv/hpm6750evkmini/hpm6750evkmini.dts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 hpmicro
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <hpmicro/hpm6750.dtsi>
+#include "hpm6750evkmini-pinctrl.dtsi"
+
+/ {
+	model = "HPMicro HPM6750EVKMINI";
+	compatible = "HPMicro,hpm6750evkmini";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,sram = &sram;
+		zephyr,itcm = &ilm;
+		zephyr,dtcm = &dlm;
+	};
+
+	core0_lm_slv: memory@1000000 {
+		device_type = "memory";
+		reg = <0x1000000 DT_SIZE_K(512)>;
+	};
+
+	core1_lm_slv: memory@1180000 {
+		device_type = "memory";
+		reg = <0x1180000 DT_SIZE_K(256)>;
+	};
+
+	noncache: memory@11c0000 {
+		device_type = "memory";
+		reg = <0x11c0000 DT_SIZE_K(256)>;
+	};
+
+	dram: memory@40000000 {
+		device_type = "memory";
+		reg = <0x40000000 DT_SIZE_M(16)>;
+	};
+
+	aliases {
+		led0 = &led_r;
+		led1 = &led_g;
+		led2 = &led_b;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led_r: led_r {
+			gpios = <&gpiob 19 GPIO_ACTIVE_LOW>;
+			label = "LEDR";
+		};
+		led_g: led_g {
+			gpios = <&gpiob 18 GPIO_ACTIVE_LOW>;
+			label = "LEDG";
+		};
+		led_b: led_b {
+			gpios = <&gpiob 20 GPIO_ACTIVE_LOW>;
+			label = "LEDB";
+		};
+	};
+};
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&pinmux_uart0>;
+	pinctrl-names = "default";
+	clock-name = <19005459>;
+	clock-src = <0>;
```

Comment:
?

Reply:
The clock name is an enumerated type defined in the hpm driver and each soc has its own header file to define it, currently it can be linked to the appropriate file correctly but I don't currently have a way to call the header file in the hal modules directly from within the dts.

Reply:
So you should probably implement a clock controller driver in zephyr.  Its unfortunate that you guys use "clock name" to mean a clock identifier.  Its going to be very confusing to use a property called `clock-name` as that has a well defined meaning in Zephyr already.

As this should just be a `clocks` property you can do something like:

`clocks = <&clk_ctrl sysctl_resource_uart0 CLK_SRC_GROUP_COMMON clock_node_uart0 0>`;

where the cells are:
- resource
- src_type
- node
- src

or:

`clocks = <&clk_ctrl clock_uart0 0>;

Where the cells are:
- name
- src